src/libgap-api.c

```diff
@@ -406,6 +407,21 @@ Obj GAP_NewPlist(Int capacity)
     return NEW_PLIST(T_PLIST_EMPTY, capacity);
 }
 
+static BOOL fitsInIntObj(Int i)
+{
+    return INT_INTOBJ_MIN <= i && i <= INT_INTOBJ_MAX;
+}
+
+Obj GAP_NewRange(Int len, Int low, Int inc)
+{
+    if (!fitsInIntObj(len)) return 0;
+    if (!fitsInIntObj(low)) return 0;
+    if (!fitsInIntObj(inc)) return 0;
```

Comment:
Do you just want to catch inc==0 here too? I don't think it's allowed.

Reply:
Good point, done